source/examples/demo-stages/CMakeLists.txt

```diff
@@ -0,0 +1,261 @@
+#
+
+# Target name
+set(target demo-stages)
+
+# Exit here if required dependencies are not met
+message(STATUS "Plugin ${target}")
+
+# Set API export file and macro
+string(MAKE_C_IDENTIFIER ${target} target_id)
+string(TOUPPER ${target_id} target_id)
+set(feature_file "include/${target}/${target}_features.h")
+set(export_file  "include/${target}/${target}_api.h")
+set(export_macro "${target_id}_API")
+
+
+#
+# Sources
+#
+
+set(include_path "${CMAKE_CURRENT_SOURCE_DIR}/include/${target}")
+set(source_path  "${CMAKE_CURRENT_SOURCE_DIR}/source")
+
+set(headers
+    ${include_path}/DemoStage.h
+    ${include_path}/DemoPipeline.h
+    ${include_path}/ShaderDemoPipeline.h
+    ${include_path}/DemoAntialiasableTriangleStage.h
+    ${include_path}/DemoAntialiasingPipeline.h
+    ${include_path}/DemoDOFCubeStage.h
+    ${include_path}/DemoDOFPipeline.h
+    ${include_path}/DemoSSAORenderingStage.h
+    ${include_path}/DemoSSAOPostprocessingStage.h
+    ${include_path}/DemoSSAOPipeline.h
+    ${include_path}/DemoTransparencyStage.h
+    ${include_path}/DemoTransparencyPipeline.h
+    ${include_path}/DemoMultiFramePipeline.h
+    ${include_path}/SpinningRectStage.h
+    ${include_path}/DemoRenderStage.h
+    ${include_path}/TimerStage.h
+    ${include_path}/ColorizeStage.h
+    ${include_path}/LightTestStage.h
+    ${include_path}/LightTestPipeline.h
+)
+
+set(sources
+    ${source_path}/plugin.cpp
+
+    ${source_path}/DemoStage.cpp
+    ${source_path}/DemoPipeline.cpp
+    ${source_path}/ShaderDemoPipeline.cpp
+    ${source_path}/DemoAntialiasableTriangleStage.cpp
+    ${source_path}/DemoAntialiasingPipeline.cpp
+    ${source_path}/DemoDOFCubeStage.cpp
+    ${source_path}/DemoDOFPipeline.cpp
+    ${source_path}/DemoSSAORenderingStage.cpp
+    ${source_path}/DemoSSAOPostprocessingStage.cpp
+    ${source_path}/DemoSSAOPipeline.cpp
+    ${source_path}/DemoTransparencyStage.cpp
+    ${source_path}/DemoTransparencyPipeline.cpp
+    ${source_path}/DemoMultiFramePipeline.cpp
+    ${source_path}/SpinningRectStage.cpp
+    ${source_path}/DemoRenderStage.cpp
+    ${source_path}/TimerStage.cpp
+    ${source_path}/ColorizeStage.cpp
+    ${source_path}/LightTestStage.cpp
+    ${source_path}/LightTestPipeline.cpp
+)
+
+# Group source files
+set(header_group "Header Files (API)")
+set(source_group "Source Files")
+source_group_by_path(${include_path} "\\\\.h$|\\\\.inl$"
+    ${header_group} ${headers})
+source_group_by_path(${source_path}  "\\\\.cpp$|\\\\.c$|\\\\.h$|\\\\.inl$"
+    ${source_group} ${sources})
+
+
+#
+# Create library
+#
+
+# Build library
+add_library(${target}
+    ${sources}
+    ${headers}
+)
+
+# Create namespaced alias
+add_library(${META_PROJECT_NAME}::${target} ALIAS ${target})
+
+# Export library for downstream projects
+export(TARGETS ${target} NAMESPACE ${META_PROJECT_NAME}:: FILE ${PROJECT_BINARY_DIR}/cmake/${target}/${target}-export.cmake)
+
+# Check for availability of module; use pre-generated version if not found
```

Comment:
If this example doesn't use any feature from the compiler feature detection, the following lines can get removed.